### ISSUE TITLE
Not training self-critical in the right way?

### ISSUE THREAD

**Issue by ruotianluo** [tgGuo15/PriorImageCaption#3]
Hi,

I look at your code and run some parts.
It seems that you are not following the original self-critical paper, where they include the <eos> when calculating the cider score.

I test your bad ending rate and bad ending removal method, my self-critical result has bad ending rate around 7%, and the removal method only decreases the cider score by 0.3.

Correct me if I were wrong.

**Comment by tgGuo15** [comment]
I calculate the cider score followed by the standard COCO tools. In my experiments for many times, the cider score drop sharply all the way. The way of calculating cider in original self-critical paper is to include 'EOS' token which is different from the previous papers.

**Comment by ruotianluo** [comment]
yes, and including eos seems to be helpful for eliminating bad ending.
